src/wazuh_modules/syscollector/syscollector_windows.c

```diff
@@ -1052,108 +1034,38 @@ void read_win_program(const char * sec_key, int arch, int root_key, int usec, co
-
-                // Increase buffer length
-
-                buffer_size += BYTEINCREMENT;
-                os_realloc(vendor, buffer_size, vendor);
-                cbData = buffer_size;
-                ret = RegQueryValueEx(program_key, "Publisher", NULL, NULL, (LPBYTE)vendor, &cbData);
-            }
-
-            if (ret == ERROR_SUCCESS && vendor[0] != '\0') {
-                cJSON_AddStringToObject(package, "vendor", vendor);
-            }
-
-            free(vendor);
+            vendor = w_reg_query_value(program_key, L"Publisher");
+            if (vendor && *vendor) cJSON_AddStringToObject(package, "vendor", vendor);
 
             // Get install date
-
-            date = (char *)malloc(TOTALBYTES);
-            cbData = buffer_size;
-
-            ret = RegQueryValueEx(program_key, "InstallDate", NULL, NULL, (LPBYTE)date, &cbData);
-            while (ret == ERROR_MORE_DATA) {
-
-                // Increase buffer length
-
-                buffer_size += BYTEINCREMENT;
-                os_realloc(date, buffer_size, date);
-                cbData = buffer_size;
-                ret = RegQueryValueEx(program_key, "InstallDate", NULL, NULL, (LPBYTE)date, &cbData);
-            }
-
-            if (ret == ERROR_SUCCESS && date[0] != '\0') {
-                cJSON_AddStringToObject(package, "install_time", date);
-            }
-
-            free(date);
+            date = w_reg_query_value(program_key, L"InstallDate");
+            if (date && *date) cJSON_AddStringToObject(package, "install_time", date);
 
             // Get install location
+            location = w_reg_query_value(program_key, L"InstallLocation");
+            if (location && *location) cJSON_AddStringToObject(package, "location", location);
 
-            location = (char *)malloc(TOTALBYTES);
-            cbData = buffer_size;
-
-            ret = RegQueryValueEx(program_key, "InstallLocation", NULL, NULL, (LPBYTE)location, &cbData);
-            while (ret == ERROR_MORE_DATA) {
-
-                // Increase buffer length
-
-                buffer_size += BYTEINCREMENT;
-                os_realloc(location, buffer_size, location);
-                cbData = buffer_size;
-                ret = RegQueryValueEx(program_key, "InstallLocation", NULL, NULL, (LPBYTE)location, &cbData);
-            }
-
-            if (ret == ERROR_SUCCESS && location[0] != '\0') {
-                cJSON_AddStringToObject(package, "location", location);
-            }
-
-            free(location);
-
-            char *string;
-            string = cJSON_PrintUnformatted(object);
+            char *string = cJSON_PrintUnformatted(object);
             mtdebug2(WM_SYS_LOGTAG, "sys_programs_windows() sending '%s'", string);
             wm_sendmsg(usec, 0, string, LOCATION, SYSCOLLECTOR_MQ);
             cJSON_Delete(object);
             free(string);
+        }
 
-        } else
-            free(program_name);
-
+        if (program_name) free(program_name);
```

Comment:
Use either the coding convention or a simple call to `free()` (since `free(NULL)` is harmless.

Reply:
Fixed in latest commit. Thanks!